launch.go

```diff
@@ -81,77 +81,101 @@ func createWorkspace(srcPaths ...string) (Workspace, error) {
 	}
 
 	w := Workspace{
-		Root:      root,
+		Root:      rootDir,
 		Src:       src,
 		Artifacts: artifacts,
 	}
 	return w, nil
 }
 
-func launch(api screwdriver.API, buildID string) error {
+func launch(api screwdriver.API, buildID string, rootDir string) error {
+	log.Printf("Fetching Build %v", buildID)
 	b, err := api.BuildFromID(buildID)
 	if err != nil {
 		return fmt.Errorf("fetching build ID %q: %v", buildID, err)
 	}
 
+	log.Printf("Fetching Job %v", b.JobID)
 	j, err := api.JobFromID(b.JobID)
 	if err != nil {
 		return fmt.Errorf("fetching Job ID %q: %v", b.JobID, err)
 	}
 
+	log.Printf("Fetching Pipeline %v", j.PipelineID)
 	p, err := api.PipelineFromID(j.PipelineID)
 	if err != nil {
 		return fmt.Errorf("fetching Pipeline ID %q: %v", j.PipelineID, err)
 	}
 
 	scm, err := parseScmURL(p.ScmURL)
-	workspace, err := createWorkspace(scm.Org, scm.Repo)
+	log.Printf("Creating Workspace in %v", rootDir)
+	w, err := createWorkspace(rootDir, scm.Org, scm.Repo)
 	if err != nil {
-		return err
+		return fmt.Errorf("creating workspace: %v", err)
 	}
 
-	err = gitClone(p.ScmURL, workspace.Src)
+	err = gitClone(p.ScmURL, w.Src)
 	if err != nil {
-		return err
+		return fmt.Errorf("cloning from git: %v", err)
```

Comment:
I don't feel like this Errorf or the one above add any information, since the error actually already says the same thing with more info. That's why we were returning nil. Perhaps log.Errorf(err) would be most appropriate here? That would exit immediately after writing the log message.